### PR TITLE
Add Interchange (IC) and Junction (JCT)

### PR DESCRIPTION
**Pull request by kried**
### Description
Feature parity, [Android implementation](https://github.com/mapbox/mapbox-navigation-android/pull/6943)

### CODE REVIEW

File: Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Foundation
+import MapboxNavigationNative
+
+/// Road object information, like interchange name.
+public struct LocalizedRoadObjectName: Equatable {
+    /// The name of the road object.
```
Comment (Udumft) [inline Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift:6]: Looks like comments are swapped

File: Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    public let language: String
```
Comment (Udumft) [inline Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift:7]: Should this property be 'languageCode' then, like it is in Foundation?

Comment (kried) [reply Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift:7]: I was looking at `language` in [RoadName](https://github.com/mapbox/mapbox-navigation-ios/blob/8d1063e0481fc7d2e9d77aa27a6d737929551d25/Sources/MapboxCoreNavigation/EHorizon/RoadName.swift#L19) also in EV.
I think I can`languageCode` in a new struct and after think about `RoadName`

**Review by Udumft — APPROVED** [review]

File: Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift (hunk 2 above)
Comment (kried) [reply Sources/MapboxCoreNavigation/EHorizon/LocalizedRoadObjectName.swift:7]: After discussion we decided to use `language` for the consistency inside the SDK

**Pull request merged**